src/uct/ib/rc/base/rc_iface.c

```diff
@@ -350,7 +350,10 @@ ucs_status_t uct_rc_iface_fc_handler(uct_rc_iface_t *iface, unsigned qp_num,
     ucs_assert(iface->config.fc_enabled);
 
     if ((ep == NULL) || (ep->flags & UCT_RC_EP_FLAG_FLUSH_CANCEL)) {
-        /* We get fc for ep which is being removed or canceled so should ignore it */
+        /* We get fc for ep which is being removed or cancelled, so should ignore it */
+        if (fc_hdr == UCT_RC_EP_FC_PURE_GRANT) {
```

Comment:
Decided to not unite this flow with the one at the bottom of `if (fc_hdr & UCT_RC_EP_FLAG_FC_GRANT) { ... }`

The reason is: here the `ep` might be `NULL`, while in the second flow we update the stats, so another `if` statement would be needed.